site_scons/prereq_tools/base.py

```diff
@@ -274,27 +274,20 @@ def run_commands(self, commands, subdir=None):
         if not self.env:
             raise Exception("PreReqComponent not initialized")
         retval = True
-        old = os.getcwd()
-        if subdir:
-            if self.__dry_run:
-                print('Would change dir to %s' % subdir)
-            else:
-                os.chdir(subdir)
 
-        print('Running commands in %s' % os.getcwd())
+        if subdir:
+            print('Running commands in {}'.format(subdir))
         for command in commands:
             command = self.env.subst(command)
             if self.__dry_run:
                 print('Would RUN: %s' % command)
                 retval = True
             else:
                 print('RUN: %s' % command)
-                if subprocess.call(command, shell=True,   # nosec
-                                   env=self.env['ENV']) != 0:
+                cmd = command.split(' ')
```

Comment:
This would not respect shell quoting to prevent space expansion would it?  I.e. if `cmd` were `ls -l "file with space"` (as a trivial example).  I think `shlex.split()` is the solution for this.